common/src/main/java/com/wynntils/wynn/model/ShamanTotemModel.java

```diff
@@ -73,7 +73,7 @@ public void onTotemSpawn(AddEntityEvent e) {
                     if (inv.size() < 4 || inv.get(3).getItem() != Items.STONE_SHOVEL) return;
 
                     int totemNumber = nextTotemSlot;
-                    nextTotemSlot = getNextTotemSlot();
```

Comment:
Actually, why don't you do like:
`int totemNumber = getNextTotemSlot();`
and let getNextTotemSlot do the cycling as well? If you want to clean things up, that is imho better. It encapsulates the handling of the totem slot into one function, which just provides a simple API: "call me and I guarantee that you will get a usable totem slot number in return!".